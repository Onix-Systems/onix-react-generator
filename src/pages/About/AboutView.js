import React from 'react';
import PropTypes from 'prop-types';
import styles from './sass/About.module.scss';

export default function AboutView({ title }) {
  return (
    <div>
      <p className={styles.appIntro}>{title}</p>
    </div>
  );
}

AboutView.propTypes = {
  title: PropTypes.string.isRequired
};
