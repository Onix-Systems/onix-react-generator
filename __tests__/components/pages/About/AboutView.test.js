import React from 'react';
import i18next from 'i18next';
import { render } from '../../../settings/test-utils';
import AboutView from '../../../../src/pages/About/AboutView';

describe('<AboutView />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = render(<AboutView title={i18next.t('aboutPage')} />);
  });

  test('should render AboutView component', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });

  test('AboutView component should have passed title', () => {
    const { getByText } = wrapper;
    expect(getByText(i18next.t('aboutPage'))).toBeInTheDocument();
  });
});
