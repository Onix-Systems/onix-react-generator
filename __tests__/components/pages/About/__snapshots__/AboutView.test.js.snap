// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutView /> should render AboutView component 1`] = `
<DocumentFragment>
  <div>
    <p
      class="appIntro"
    >
      About page
    </p>
  </div>
</DocumentFragment>
`;
